src/test/kotlin/KoverPluginTest.kt

```diff
@@ -0,0 +1,46 @@
+package kotlinx.kover.json
+
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import java.nio.file.Files
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+
+class KoverPluginTest {
```

Comment:
I found the existing tests pretty difficult to understand and unpick. I created this small test so I could more easily adjust the test to fetch the output build log.

Reply:
Since this test does not test the functionality of the plugin and is used for local debugging, I think it is better to leave it in the local branch.
In addition, it slightly smears the testing levels across two modules - in this project there are  partial unit-tests, and not for the entire plugin, the dependency in it on `gradleTestKit()` is redundant.

In addition, remember that the functional tests run by Gradle itself are not entirely honest, because there are minor differences in the environment and dependencies. Therefore, for verification, I recommend publishing to the local repository `gradlew publishToMavenLocal` and already from `mavenLocal()` use the plugin.

Reply:
It tests whether the Kover plugin can be applied, that's functionality :)

> Therefore, for verification, I recommend publishing to the local repository `gradlew publishToMavenLocal` and already from `mavenLocal()` use the plugin.

That's a lot more work than running a single unit test!

I found this test really useful when trying to track down why the other tests weren't working. The existing tests didn't give enough detail. I did try adjusting them, but I didn't really understand how they're set up and created. I find it much easier to see the build and settings files under test, and adjust the command and output log.

Reply:
Each of the available functional tests checks that the plugin is being applied, a separate test for this is useless.

In any case, example project launches will be added soon (see #219), so there is no need for a separate such test, especially in another module.

Reply:
The best way is to improve the error messages of the existing functional tests (adding the error log and the contents of the build script to the message, I will do it after merge #219), rather than adding a completely duplicate test.